src/MongoDB.AspNetCore.OData/MongoExpressionRewriter.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace MongoDB.AspNetCore.OData;
+
+public class MongoExpressionRewriter : ExpressionVisitor
```

Comment:
It could be internal class. Benefits of having it internal that we are free to change implementation of internal classes without worrying about API backward compatibility.

Reply:
Done.